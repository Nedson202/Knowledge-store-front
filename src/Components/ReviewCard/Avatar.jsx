import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';

class Avatar extends Component {
  renderInitials(user) {
    const matches = user.match(/\b(\w)/g);
    const acronym = matches.join('');
    const { color } = this.props;
    return (
      <div
        className="avatar-placeholder text-capitalize"
        style={{ backgroundColor: color }}
      >
        {acronym}

      </div>
    );
  }

  render() {
    const { user } = this.props;
    return (
      <Fragment>
        {this.renderInitials(user)}
      </Fragment>
    );
  }
}

Avatar.propTypes = {
  user: PropTypes.string,
  color: PropTypes.string
};

Avatar.defaultProps = {
  user: '',
  color: ''
};

export default Avatar;
