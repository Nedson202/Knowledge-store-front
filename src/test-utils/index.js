import React from 'react';
import { MockedProvider } from 'react-apollo/test-utils';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../redux';

// eslint-disable-next-line react/prop-types
export const AllProviders = ({ children, customStore }) => (
  <MockedProvider removeTypename>
    <Provider store={customStore || store}>
      <BrowserRouter>
        {children}
      </BrowserRouter>
    </Provider>
  </MockedProvider>
);

export * from '@testing-library/react';
