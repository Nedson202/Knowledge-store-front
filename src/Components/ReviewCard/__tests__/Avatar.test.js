import React from 'react';
import { render } from 'test-utils';

import Avatar from '../Avatar';

const mockProps = {
  user: 'Cooper AL',
  color: 'green',
};

test('Avatar component should render', () => {
  const { getByText, asFragment, } = render(<Avatar {...mockProps} />);

  getByText('CA');

  expect(asFragment()).toMatchSnapshot();
});
