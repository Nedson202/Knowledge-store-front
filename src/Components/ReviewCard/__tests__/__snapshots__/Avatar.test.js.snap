// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render 1`] = `
<DocumentFragment>
  <div
    class="avatar-placeholder text-capitalize"
    style="background-color: green;"
  >
    CA
  </div>
</DocumentFragment>
`;
